R/utils.R

```diff
@@ -120,29 +124,82 @@ expose_stan_fns <- function(files, target_dir, ...) {
 #'
 #' # Load the script used to generate these examples
 #' # Optionally source this script to regenerate the example
-#' readLines(enw_example(type = "script"))
-enw_example <- function(type = "nowcast") {
-  type <- match.arg(
-    type,
-    choices = c(
-      "nowcast", "preprocessed_observations", "observations", "script"
-    )
-  )
+#' readLines(enw_example(type = "scripts"))
+enw_example <- function(
+  type = c("nowcast", "preprocessed_observations", "observations", "scripts")
+) {
+  type <- match.arg(type)
 
   if (type %in% c("nowcast", "preprocessed_observations", "observations")) {
-    file <- system.file("extdata", paste0(type, ".rds"), package = "epinowcast")
-  } else if (type %in% "script") {
-    file <- system.file("scripts", "germany_example.R", package = "epinowcast")
+    return(readRDS(
+      system.file("extdata", sprintf("%s.rds", type), package = "epinowcast")
+    ))
+  } else if (type == "scripts") {
+    return(list.files(
+      system.file(type, package = "epinowcast"),
+      full.names = TRUE)
+    )
   }
 
-  if (type %in% "script") {
-    out <- file
+}
+
+#' @title Coerce Dates
+#'
+#' @description Provides consistent coercion of inputs to [IDate]
+#' with error handling
+#'
+#' @param dates A vector-like input, which the function attempts
+#' to coerce via [data.table::as.IDate()].
+#'
+#' @return An [IDate] vector.
+#'
+#' @details If any of the elements of `dates` cannot be coerced,
+#' this function will result in an error, indicating all indices
+#' which cannot be coerced to [IDate].
+#'
+#' Internal methods of [epinowcast] assume dates are represented
+#' as [IDate].
+#'
+#' @export
+#' @importFrom data.table as.IDate
+#' @family utils
+#' @examples
+#' # works
+#' coerce_date(c("2020-05-28", "2020-05-29"))
+#' # does not, indicates index 2 is problem
+#' tryCatch(
+#'  coerce_date(c("2020-05-28", "2020-o5-29")),
+#'  error = function(e) print(e)
```

Comment:
style

Reply:
apparently styler doesn't reach into examples, so adding that as an n.b. in CONTRIBUTING